common/src/main/java/com/hedera/block/common/utils/Preconditions.java

```diff
@@ -75,6 +75,65 @@ public static int requirePositive(final int toCheck) {
         return requirePositive(toCheck, null);
     }
 
+    /**
+     * This method asserts a given integer is a positive. An integer is positive
+     * if it is NOT equal to zero and is greater than zero.
+     *
+     * @param toCheck the integer to check if it is a positive power of two
+     * @param errorMessage the error message to be used in the exception if the
+     * input integer to check is not positive, if null, a default message will
+     * be used
+     * @return the number to check if it is positive
+     * @throws IllegalArgumentException if the input number to check is not
+     * positive
+     */
+    public static int requirePositive(final int toCheck, final String errorMessage) {
+        if (0 >= toCheck) {
+            final String message = Objects.isNull(errorMessage)
+                    ? "The input integer [%d] is required be positive.".formatted(toCheck)
+                    : errorMessage;
```

Comment:
That indeed looks cleaner, true. My though was the following:
It would be better to not allow the caller to make a mistake (supply a null String) since this is a runtime exception and it could blow with a null pointer, instead of the actual intended exception due to not passing the precondition. Granted, the stack trace will eventually lead you to that problem, but you have to find and solve a problem just to reach the actual problem.
I have had such troubles in the past, initially, you see a null pointer and start looking for a problem there, putting some time into that. Which is in practice misleading.

I would suggest that we allow nulls here, we will only touch the ternary in cases of actual failure of the precondition. And we eliminate the possibility of a null pointer.